src/lifecycle.py

```diff
@@ -24,8 +30,75 @@ def stop(stackname, *services):
 @task
 @requires_aws_stack
 @timeit
+@echo_output
 def restart(stackname):
-    lifecycle.restart(stackname)
+    return lifecycle.restart(stackname)
+
+
+@task
+def restart_all(statefile):
+    "restarts all running ec2 instances. multiple nodes are restarted serially and failures prevent the rest of the node from being restarted"
+
+    os.system("touch " + statefile)
+
+    results = core.active_stack_names(core.find_region())
+
+    u1404 = [
+        'api-gateway',
+        'journal',
+        'search',
+        'api-dummy',
+        'medium',
+    ]
+
+    legacy = [
+        'elife-api'
+    ]
+
+    dont_do = u1404 + legacy
+
+    # order not preserved
+    do_first = [
+        'master-server',
+        'bus',
+        'elife-alfred',
+        'elife-bot',
+        'iiif',
+    ]
+
+    pname = lambda stackname: core.parse_stackname(stackname)[0]
+    todo = sorted(results, key=pname in do_first, reverse=True)
+    todo = filter(lambda stackname: pname(stackname) not in dont_do, todo)
+
+    print('todo:')
+    print(json.dumps(todo, indent=4))
+
+    with open(statefile, 'r') as fh:
+        done = fh.read().split('\n')
+
+    with open(statefile, 'a') as fh:
+        print('writing state to', fh.name)
```

Comment:
would use a logger over `print` in general, even after closing a terminal an audit remains in the log file just in case

Reply:
done